src/cli/format.js

```diff
@@ -312,9 +312,11 @@ async function formatFiles(context) {
       );
       process.exit(2);
     }
-    const stat = await statSafe(cacheFilePath);
-    if (stat) {
-      await fs.unlink(cacheFilePath);
+    if (!context.argv.cacheLocation) {
+      const stat = await statSafe(cacheFilePath);
+      if (stat) {
+        await fs.unlink(cacheFilePath);
+      }
```

Comment:
If I didn't get this wrong, this line is going to remove default cache file when running Prettier without `--cache`, but I don't think we should.

Reply:
This is a direct implementation of ESLint's behavior and has nothing to do with --cache-location. It will be fixed in another PR.